### PR TITLE
Return Bool when appropriate

### PR DESCRIPTION
**Pull request by sheaf**
Some of the backend functions return a boolean, which was being discarded.

### CODE REVIEW

**Comment by sheaf** [comment]
I also updated the main functions to use `(/=0)` instead of `(==1)`.

File: src/DearImGui/SDL/OpenGL.hs [hunk 1]
```diff
@@ -49,9 +49,9 @@ Cpp.using "namespace ImGui"
 
 
 -- | Wraps @ImGui_ImplSDL2_InitForOpenGL@.
-sdl2InitForOpenGL :: MonadIO m => Window -> GLContext -> m ()
-sdl2InitForOpenGL (Window windowPtr) glContext = liftIO do
-  [C.exp| void { ImGui_ImplSDL2_InitForOpenGL((SDL_Window*)$(void* windowPtr), $(void* glContextPtr)); } |]
+sdl2InitForOpenGL :: MonadIO m => Window -> GLContext -> m Bool
+sdl2InitForOpenGL (Window windowPtr) glContext = ( 0 /= ) <$> liftIO do
+  [C.exp| bool { ImGui_ImplSDL2_InitForOpenGL((SDL_Window*)$(void* windowPtr), $(void* glContextPtr)) } |]
```
Comment (ocharles) [inline src/DearImGui/SDL/OpenGL.hs:54]: I like to keep the `liftIO` right at the top, so for consistency:

```suggestion
sdl2InitForOpenGL (Window windowPtr) glContext = liftIO do
  ( 0 /= ) <$> [C.exp| bool { ImGui_ImplSDL2_InitForOpenGL((SDL_Window*)$(void* windowPtr), $(void* glContextPtr)) } |]
```

Comment (sheaf) [reply src/DearImGui/SDL/OpenGL.hs]: OK

File: src/DearImGui/OpenGL.hs [hunk 2]
```diff
@@ -42,9 +42,9 @@ Cpp.using "namespace ImGui"
 
 
 -- | Wraps @ImGui_ImplOpenGL2_Init@.
-openGL2Init :: MonadIO m => m ()
-openGL2Init = liftIO do
-  [C.exp| void { ImGui_ImplOpenGL2_Init(); } |]
+openGL2Init :: MonadIO m => m Bool
+openGL2Init = ( 0 /= ) <$> liftIO do
+  [C.exp| bool { ImGui_ImplOpenGL2_Init() } |]
```
Comment (ocharles) [inline src/DearImGui/OpenGL.hs:47]:
```suggestion
openGL2Init :: MonadIO m => m Bool
openGL2Init = liftIO do
  ( 0 /= ) <$> [C.exp| bool { ImGui_ImplOpenGL2_Init() } |]
```

**Comment by ocharles** [comment]
Thanks! :factory_worker:

**Pull request merged**